javaslang/src/main/java/javaslang/collection/CharSeq.java

```diff
@@ -2159,10 +2160,14 @@ public CharSeq replace(CharSequence target, CharSequence replacement) {
      * @return the Seq of strings computed by splitting this string
      * around matches of the given regular expression
      * @throws PatternSyntaxException if the regular expression's syntax is invalid
-     * @see java.util.regex.Pattern
+     * @see Pattern
      */
-    public Seq<CharSeq> split(String regex, int limit) {
-        Array<String> split = Array.wrap(back.split(regex, limit));
+    @Deprecated(/* Use splitSeq instead, will be removed in 3.0.0 */)
+    public CharSeq[] split(String regex, int limit) {
+        return splitSeq(regex, limit).toJavaArray(CharSeq.class);
+    }
+    public Seq<CharSeq> splitSeq(String regex, int limit) {
+        final Seq<String> split = Array.wrap(back.split(regex, limit));
```

Comment:
No, we do not need to worry, this is how Java 8's impl of `java.lang.String.split` works:

```java
    public String[] split(String regex, int limit) {
        /* fastpath if the regex is a
         (1)one-char String and this character is not one of the
            RegEx's meta characters ".$|()[{^?*+\\", or
         (2)two-char String and the first char is the backslash and
            the second is not the ascii digit or ascii letter.
         */
        char ch = 0;
        if (((regex.value.length == 1 &&
             ".$|()[{^?*+\\".indexOf(ch = regex.charAt(0)) == -1) ||
             (regex.length() == 2 &&
              regex.charAt(0) == '\\' &&
              (((ch = regex.charAt(1))-'0')|('9'-ch)) < 0 &&
              ((ch-'a')|('z'-ch)) < 0 &&
              ((ch-'A')|('Z'-ch)) < 0)) &&
            (ch < Character.MIN_HIGH_SURROGATE ||
             ch > Character.MAX_LOW_SURROGATE))
        {
            int off = 0;
            int next = 0;
            boolean limited = limit > 0;
            ArrayList<String> list = new ArrayList<>();
            while ((next = indexOf(ch, off)) != -1) {
                if (!limited || list.size() < limit - 1) {
                    list.add(substring(off, next));
                    off = next + 1;
                } else {    // last one
                    //assert (list.size() == limit - 1);
                    list.add(substring(off, value.length));
                    off = value.length;
                    break;
                }
            }
            // If no match was found, return this
            if (off == 0)
                return new String[]{this};

            // Add remaining segment
            if (!limited || list.size() < limit)
                list.add(substring(off, value.length));

            // Construct result
            int resultSize = list.size();
            if (limit == 0) {
                while (resultSize > 0 && list.get(resultSize - 1).length() == 0) {
                    resultSize--;
                }
            }
            String[] result = new String[resultSize];
            return list.subList(0, resultSize).toArray(result);
        }
        return Pattern.compile(regex).split(this, limit);
    }
```

(example taken from Java's original source code)